pkg/smartcontract/parameter.go

```diff
@@ -205,7 +225,11 @@ func (p *Parameter) EncodeBinary(w *io.BinWriter) {
 	case StringType:
 		w.WriteString(p.Value.(string))
 	case IntegerType:
-		w.WriteU64LE(uint64(p.Value.(int64)))
+		val, ok := p.Value.(*big.Int)
```

Comment:
Who is encoding `smartcontract.Parameter` into binary?

Reply:
Nobody, it seems.